gprofiler/profilers/java.py

```diff
@@ -279,6 +280,72 @@ def get_ap_version() -> str:
     return Path(resource_path("java/async-profiler-version")).read_text()
 
 
+@functools.lru_cache(maxsize=1024)
+def get_jvm_flags(
+    process: Process,
+) -> List[Tuple[Optional[str], Optional[str], Optional[str], str, Optional[List[str]]]]:
+    return [
+        (flag_name, flag_type, flag_value, flag_source, flag_kind)
+        for flag_name, flag_type, flag_value, flag_source, flag_kind in get_jvm_flags_raw(process)
+        if filter_jvm_flags(flag_name, flag_type, flag_value, flag_source, flag_kind) is True
+    ]
+
+
+def filter_jvm_flags(
+    _flag_name: Optional[str],
+    flag_type: Optional[str],
+    _flag_value: Optional[str],
+    flag_source: str,
+    flag_kind: Optional[List[str]],
+) -> bool:
+    if flag_source == "default":
+        return False
+
+    if flag_type and flag_type not in ["bool", "int", "uint", "intx", "uintx", "uint64_t", "size_t", "double"]:
```

Comment:
Why would `flag_type` be None?  It doesn't appear to be the case in your parsing

Reply:
`flag_type` was a regex match so it could be None, but Im changing it now as I wrote in https://github.com/Granulate/gprofiler/pull/660#discussion_r1085408196